### PR TITLE
enable haproxy

### PR DESCRIPTION
**Pull request by perryjrandall**
Enable the proper haproxy config for fullnode / validator docker compose

Unify on all configs, fix comments, add testsuite 

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4965)

### CODE REVIEW

File: docker/compose/aptos-node/blocked.ips [hunk 1]
```diff
@@ -0,0 +1 @@
+# Add Blocked Ips here
```
Comment (sherry-x) [inline docker/compose/aptos-node/blocked.ips:1]: can you add some example format here?

**Review by sherry-x — APPROVED** [review]

File: docker/compose/aptos-node/haproxy.cfg [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    stick-table type ip size 128K expire 30m store gpc1,conn_rate(1m),bytes_out_cnt	##about 500MB of memory
+    tcp-request connection track-sc0 src 						   #update table with src ip as key, store in sc0
+
+    #We Count rate-limit manualy -- Will be more CPU intensieve but will allow whitelists to enter and up to rateLimitSession non blacklisted IPs.
+    tcp-request connection track-sc1 int(1) table CONN_RATE
+
+    #This connection is silently dropped no reason to count it for rateLimitSession
+    tcp-request connection sc-inc-gpc1(1) unless ip_high_conn_rate
+
+    # an IP that was blacklisted due to to many unsucsessfull tcp attempts
```
Comment (perryjrandall) [inline docker/compose/aptos-node/haproxy.cfg:56]: update comment

File: docker/compose/aptos-node/haproxy.cfg [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    stick-table type ip size 128K expire 30m store gpc0,gpc1,conn_rate(1m),bytes_out_cnt	##about 500MB of memory
+    tcp-request connection track-sc0 src 						   #update table with src ip as key, store in sc0
+
+    #We Count rate-limit manualy -- Will be more CPU intensieve but will allow whitelists to enter and up to rateLimitSession non blacklisted IPs.
+    tcp-request connection track-sc1 int(1) table CONN_RATE
+
+    #This connection is silently dropped no reason to count it for rateLimitSession
+    tcp-request connection sc-inc-gpc1(1) unless ip_high_conn_rate
+
+    # an IP that was blacklisted due to to many unsucsessfull tcp attempts
```
Comment (perryjrandall) [inline docker/compose/aptos-node/haproxy.cfg:95]: update comment

File: docker/compose/aptos-node/docker-compose.yaml [hunk 4]
```diff
@@ -3,6 +3,37 @@
...
+      # Validator network
+      - 6180
+      # Fullnode access to validators
+      - 6181
+      # Validator Metrics
+      - 9101
+      # Haproxy internal stats
+      - 9102
+      # Disable access to rest API port 80 for validator by default
+      # - 80
```
Comment (perryjrandall) [inline docker/compose/aptos-node/docker-compose.yaml:27]: 8180

File: terraform/helm/aptos-node/files/haproxy.cfg [hunk 5]
```diff
@@ -1,51 +1,150 @@
...
+    tcp-request connection track-sc0 src 						   #update table with src ip as key, store in sc0
+
+    #We Count rate-limit manualy -- Will be more CPU intensieve but will allow whitelists to enter and up to rateLimitSession non blacklisted IPs.
+    tcp-request connection track-sc1 int(1) table CONN_RATE
+
+    #This connection is silently dropped no reason to count it for rateLimitSession
+    tcp-request connection sc-inc-gpc1(1) unless ip_high_conn_rate
+
+    # an IP that was blacklisted due to to many unsucsessfull tcp attempts
+    #-1- Enforece Blacklist
```
Comment (perryjrandall) [inline terraform/helm/aptos-node/files/haproxy.cfg:55]: update comment

File: terraform/helm/aptos-node/files/haproxy.cfg [hunk 6]
```diff
@@ -1,51 +1,150 @@
...
```
Comment (perryjrandall) [inline terraform/helm/aptos-node/files/haproxy.cfg:93]: update comment

File: docker/compose/aptos-node/haproxy-fullnode.cfg [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+    # enough for 1 successfull + 5 unsuccessfull HB(10 sec interval) + 20 sec timeout
+    timeout server 80s
+    timeout client 80s
+
+    timeout client-fin 3s #How long to hold an interrupted client connection.
+    timeout server-fin 1s
+
+frontend fullnode-fn
+    mode tcp
+    bind :6182
```
Comment (perryjrandall) [inline docker/compose/aptos-node/haproxy-fullnode.cfg:38]: add same config for 6182

File: terraform/helm/aptos-node/files/haproxy.cfg [hunk 8]
```diff
@@ -1,51 +1,147 @@
...
+    tcp-request connection track-sc0 src 						   #update table with src ip as key, store in sc0
+
+    #We Count rate-limit manualy -- Will be more CPU intensieve but will allow whitelists to enter and up to rateLimitSession non blacklisted IPs.
+    tcp-request connection track-sc1 int(1) table CONN_RATE
+
+    #This connection is silently dropped no reason to count it for rateLimitSession
+    tcp-request connection sc-inc-gpc1(1) unless ip_high_conn_rate
+
+    # an IP is rejected due to to many unsucsessfull tcp attempts
+    #-1- Enforece Blacklist
```
Comment (perryjrandall) [inline terraform/helm/aptos-node/files/haproxy.cfg:55]: remove comment

File: docker/compose/aptos-node/haproxy-fullnode.cfg [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+    tcp-request connection track-sc0 src 						   #update table with src ip as key, store in sc0
+
+    #We Count rate-limit manualy -- Will be more CPU intensieve but will allow whitelists to enter and up to rateLimitSession non blacklisted IPs.
+    tcp-request connection track-sc1 int(1) table CONN_RATE
+
+    #This connection is silently dropped no reason to count it for rateLimitSession
+    tcp-request connection sc-inc-gpc1(1) unless ip_high_conn_rate
+
+    # an IP that was blacklisted due to to many unsucsessfull tcp attempts
+    #-1- Enforece Blacklist
```
Comment (perryjrandall) [inline docker/compose/aptos-node/haproxy-fullnode.cfg:57]: remove

File: docker/compose/aptos-node/haproxy-fullnode.cfg [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+    tcp-request connection silent-drop if ip_high_conn_rate
+
+    #an IP that had a sucessfull connection.
+    #-2- Allow Whitelist
+    tcp-request connection accept if { sc0_get_gpc1() ge 1 }
+
+    #-3- Enforece RateLimit. Connection attempts by *new* IPs/sec
```
Comment (perryjrandall) [inline docker/compose/aptos-node/haproxy-fullnode.cfg:64]: typo, enforce

File: docker/compose/aptos-node/haproxy.cfg [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+    tcp-request connection track-sc0 src 						   #update table with src ip as key, store in sc0
+
+    #We Count rate-limit manualy -- Will be more CPU intensieve but will allow whitelists to enter and up to rateLimitSession non blacklisted IPs.
+    tcp-request connection track-sc1 int(1) table CONN_RATE
+
+    #This connection is silently dropped no reason to count it for rateLimitSession
+    tcp-request connection sc-inc-gpc1(1) unless ip_high_conn_rate
+
+    # an IP is rejected due to to many unsucsessfull tcp attempts
+    #-1- Enforece Blacklist
```
Comment (perryjrandall) [inline docker/compose/aptos-node/haproxy.cfg:57]: remove

File: docker/compose/aptos-node/haproxy.cfg [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+    tcp-request connection silent-drop if ip_high_conn_rate
+
+    #an IP that had a sucessfull connection.
+    #-2- Allow Whitelist
+    tcp-request connection accept if { sc0_get_gpc1() ge 1 }
+
+    #-3- Enforece RateLimit. Connection attempts by *new* IPs/sec
```
Comment (perryjrandall) [inline docker/compose/aptos-node/haproxy.cfg:64]: typo, enforce

File: docker/compose/aptos-node/haproxy.cfg [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+    tcp-request connection track-sc0 src 						   #update table with src ip as key, store in sc0
+
+    #We Count rate-limit manualy -- Will be more CPU intensieve but will allow whitelists to enter and up to rateLimitSession non blacklisted IPs.
+    tcp-request connection track-sc1 int(1) table CONN_RATE
+
+    #This connection is silently dropped no reason to count it for rateLimitSession
+    tcp-request connection sc-inc-gpc1(1) unless ip_high_conn_rate
+
+    # an IP is rejected due to to many unsucsessfull tcp attempts
+    #-1- Enforece Blacklist
```
Comment (perryjrandall) [inline docker/compose/aptos-node/haproxy.cfg:96]: remove

File: docker/compose/aptos-node/haproxy.cfg [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+    tcp-request connection silent-drop if ip_high_conn_rate
+
+    #an IP that had a sucessfull connection.
+    #-2- Allow Whitelist
+    tcp-request connection accept if { sc0_get_gpc1() ge 1 }
+
+    #-3- Enforece RateLimit. Connection attempts by *new* IPs/sec
```
Comment (perryjrandall) [inline docker/compose/aptos-node/haproxy.cfg:103]: typo, enforce

File: terraform/helm/aptos-node/files/haproxy.cfg [hunk 15]
```diff
@@ -1,51 +1,147 @@
...
+    tcp-request connection silent-drop if ip_high_conn_rate
+
+    #an IP that had a sucessfull connection.
+    #-2- Allow Whitelist
+    tcp-request connection accept if { sc0_get_gpc1() ge 1 }
+
+    #-3- Enforece RateLimit
```
Comment (perryjrandall) [inline terraform/helm/aptos-node/files/haproxy.cfg:62]: typo, enforce

File: terraform/helm/aptos-node/files/haproxy.cfg (hunk 6 above)
Comment (perryjrandall) [reply terraform/helm/aptos-node/files/haproxy.cfg:93]: remove comment

File: terraform/helm/aptos-node/files/haproxy.cfg [hunk 16]
```diff
@@ -1,51 +1,147 @@
...
```
Comment (perryjrandall) [inline terraform/helm/aptos-node/files/haproxy.cfg:100]: typo, enforce

File: terraform/helm/aptos-node/files/haproxy.cfg [hunk 17]
```diff
@@ -76,6 +172,33 @@ frontend {{ $config.name }}-aptosnet
...
+    tcp-request connection track-sc0 src 						   #update table with src ip as key, store in sc0
+
+    #We Count rate-limit manualy -- Will be more CPU intensieve but will allow whitelists to enter and up to rateLimitSession non blacklisted IPs.
+    tcp-request connection track-sc1 int(1) table CONN_RATE
+
+    #This connection is silently dropped no reason to count it for rateLimitSession
+    tcp-request connection sc-inc-gpc1(1) unless ip_high_conn_rate
+
+    # an IP is rejected due to to many unsucsessfull tcp attempts
+    #-1- Enforece Blacklist
```
Comment (perryjrandall) [inline terraform/helm/aptos-node/files/haproxy.cfg:187]: remove comment

File: terraform/helm/aptos-node/files/haproxy.cfg [hunk 18]
```diff
@@ -76,6 +172,33 @@ frontend {{ $config.name }}-aptosnet
...
+    tcp-request connection silent-drop if ip_high_conn_rate
+
+    #an IP that had a sucessfull connection.
+    #-2- Allow Whitelist
+    tcp-request connection accept if { sc0_get_gpc1() ge 1 }
+
+    #-3- Enforece RateLimit
```
Comment (perryjrandall) [inline terraform/helm/aptos-node/files/haproxy.cfg:194]: typo, enforce

**Review by perryjrandall — COMMENTED** [review]
update comments and lets goooo

File: docker/compose/aptos-node/haproxy-fullnode.cfg [hunk 19]
```diff
@@ -0,0 +1,134 @@
...
+    stick-table type ip size 128K expire 30m store gpc1,conn_rate(1m),bytes_out_cnt	##about 500MB of memory
+    tcp-request connection track-sc0 src 						   #update table with src ip as key, store in sc0
+
+    #We Count rate-limit manualy -- Will be more CPU intensieve but will allow whitelists to enter and up to rateLimitSession non blacklisted IPs.
+    tcp-request connection track-sc1 int(1) table CONN_RATE
+
+    #This connection is silently dropped no reason to count it for rateLimitSession
+    tcp-request connection sc-inc-gpc1(1) unless ip_high_conn_rate
+
+    # an IP that was blacklisted due to to many unsucsessfull tcp attempts
```
Comment (perryjrandall) [inline docker/compose/aptos-node/haproxy-fullnode.cfg:56]: update?

**Pull request merged**